src/middlewared/middlewared/alert/source/pools.py

```diff
@@ -1,30 +1,52 @@
-from middlewared.alert.base import AlertClass, AlertCategory, AlertLevel, Alert, AlertSource
-from middlewared.alert.schedule import CrontabSchedule
+from middlewared.alert.base import AlertClass, AlertCategory, AlertLevel, Alert, OneShotAlertClass
 
 
-class PoolUSBDisksAlertClass(AlertClass):
+class PoolUSBDisksAlertClass(AlertClass, OneShotAlertClass):
     category = AlertCategory.STORAGE
     level = AlertLevel.WARNING
     title = 'Pool consuming USB disks'
     text = '%(pool)r is consuming USB devices %(disks)r which is not recommended.'
 
+    async def get_usb_disks(self, pool_name, disks):
+        try:
+            return [disk for disk in filter(
+                lambda d: d in disks and disks[d]['bus'] == 'USB',
+                await self.middleware.call('zfs.pool.get_disks', pool_name)
+            )]
+        except Exception:
+            return []
 
-class PoolDisksChecksAlertSource(AlertSource):
-    schedule = CrontabSchedule(hour=0)  # every 24 hours
-    run_on_backup_node = False
+    async def create(self, args):
+        pool_name = args['pool_name']
+        disks = args['disks']
 
-    async def check(self):
-        alerts = []
+        if (usb_disks := await self.get_usb_disks(pool_name, disks)):
+            return Alert(PoolUSBDisksAlertClass, {'pool': pool_name, 'disks': ', '.join(usb_disks)}, key=pool_name)
 
-        for pool in filter(
-            lambda p: p['state'] == 'ONLINE',
-            (await self.middleware.call('zfs.pool.query_imported_fast')).values()
-        ):
-            usb_disks = await self.middleware.call('pool.get_usb_disks', pool['name'])
-            if usb_disks:
-                alerts.append(Alert(
-                    PoolUSBDisksAlertClass,
-                    {'pool': pool['name'], 'disks': ', '.join(usb_disks)},
-                ))
+    async def delete(self, alerts, query):
+        return list(filter(lambda x: x.key != query, alerts))
 
-        return alerts
+
+class PoolUpgradedAlertClass(AlertClass, OneShotAlertClass):
+    category = AlertCategory.STORAGE
+    level = AlertLevel.WARNING
+    title = "New Feature Flags Are Available for Pool"
+    text = (
+        "New ZFS version or feature flags are available for pool '%s'. Upgrading pools is a one-time process that can "
+        "prevent rolling the system back to an earlier TrueNAS version. It is recommended to read the TrueNAS release "
+        "notes and confirm you need the new ZFS feature flags before upgrading a pool."
+    )
+
+    async def is_upgraded(self, pool_name):
+        try:
+            return await self.middleware.call('zfs.pool.is_upgraded', pool_name)
+        except Exception:
+            return False
+
+    async def create(self, args):
+        pool = args['pool_name']
+        if await self.is_upgraded(pool):
```

Comment:
Isn't a pool "upgraded" if it has all the feature flags enabled? (and we generally want all pools to be upgraded and we create alerts if they are not upgraded)?

Reply:
oops! Fixed :)